src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -287,7 +287,7 @@ public function denormalize($data, $class, $format = null, array $context = [])
             try {
                 $this->setAttributeValue($object, $attribute, $value, $format, $context);
             } catch (InvalidArgumentException $e) {
-                throw new NotNormalizableValueException($e->getMessage(), $e->getCode(), $e);
+                throw new NotNormalizableValueException(sprintf('%s. The type of the "%s" attribute for class "%s" is wrong ("%s" given).', $e->getMessage(), $attribute, $class, \gettype($value)), $e->getCode(), $e);
```

Comment:
Not sure we should mention "The type of" nor the type of `$value`. This is already present in the original exception message (`Expected argument of type "%s", "%s" given`).
Should rather be something like: `Failed to denormalize attribute "%s" value for class "%s": %originalMessage%`

The aim here is to add the attribute and class info to the original exception. That's all we need to do.